### ISSUE TITLE
`--prefer-offline` should not act like `--offline`

### ISSUE THREAD

**Issue by Spy-Seth** [yarnpkg/yarn#2173]
**Do you want to request a *feature* or report a *bug*?**

Bug

**What is the current behavior?**

I use the offline cache feature of yarn. In my `.yarnrc` I have this:
 
```yaml
yarn-offline-mirror "/var/cache/yarn-offline-cache"
```
My lock file if regenerated to use the offline cache.

If I try `$ yarn install`  with an empty `/var/cache/yarn-offline-cache` directory, my dependencies are downloaded and cached: perfect.

If I try `$ yarn install --offline`  with an empty `/var/cache/yarn-offline-cache` directory, my dependencies are not downloaded and the install faild: perfect. If I do the same with the directory populated, nothing to download so it's works.

But if I try `$ yarn install --prefer-offline`  with an empty `/var/cache/yarn-offline-cache` directory, my dependencies are not downloaded: Oups! Not what that is expected. 

Like the options said, if no package available in the cache it should download it from the distant repository. For now the option is acting like the `--offline` one, that not what we should expect.

**Please mention your node.js, yarn and operating system version.**

yarn v0.17.10

**Comment by danpalmer** [comment]
Any movement on this? I've just run `yarn install --prefer-offline` and I get:

```
verbose 2.206 Error: Couldn't find package "@kadira/storybook" on the "npm" registry.
    at /usr/lib/node_modules/yarn/lib/resolvers/registries/npm-resolver.js:209:15
    at Generator.next (<anonymous>)
    at step (/usr/lib/node_modules/yarn/node_modules/babel-runtime/helpers/asyncToGenerator.js:17:30)
    at /usr/lib/node_modules/yarn/node_modules/babel-runtime/helpers/asyncToGenerator.js:28:13
error Couldn't find package "@kadira/storybook" on the "npm" registry.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```

**Comment by BlueRaja** [comment]
Probably related to https://github.com/yarnpkg/yarn/issues/1482

**Comment by bestander** [comment]
This is fixed in 0.25
